src/Main.elm

```diff
@@ -227,6 +225,78 @@ getPair orphan allCards =
     orphan :: List.filter (\card -> card.match == orphan.value) allCards
 
 
+type Streak
+    = Correct Int
+    | Incorrect Int
+
+
+updateStreak : Card -> Model -> Streak
+updateStreak selectedCard model =
+    case model.streak of
+        Correct tries ->
+            if matched model.cards selectedCard then
+                Correct (tries + 1)
+
+            else
+                Incorrect 1
+
+        Incorrect tries ->
+            if matched model.cards selectedCard then
+                Correct 1
+
+            else
+                Incorrect (tries + 1)
+
+
+needsProgressReport : Card -> Model -> ( Bool, String )
+needsProgressReport selectedCard model =
+    let
+        progressType : ProgressTrigger
+        progressType =
+            -- Only one new card turned over
+            if cardsIsEven (revealedCards model.cards) then
+                None
+
+            else if notMatched model.cards selectedCard then
+                -- This is an incorrect try check if 3 in a row
+                case model.streak of
+                    Incorrect tries ->
+                        if modBy 3 (tries + 1) == 0 then
+                            MissedThreeInARow
+
+                        else
+                            None
+
+                    Correct _ ->
+                        None
+
+            else
+            -- This is a correct try
+            -- Check if this is the final pair
+            if
+                List.length model.cards == (List.length (revealedCards model.cards) + 1)
+            then
+                MatchedAll
+
+            else if
+                modBy 3 (List.length (revealedCards model.cards) + 1)
+                    == 0
+                -- Check if total revealed multiple of 3
+            then
+                MatchedThree
+
+            else
+                None
+    in
+    if progressType == None then
+        ( False, "" )
+
+    else
+        ( True
+        , Maybe.withDefault "" (List.head (getProgressSpeech progressType))
```

Comment:
I think this is always picking the first item from the list of options in each category?

Reply:
Yup. It's a WIP. Good spot.